src/Microsoft.PowerShell.Commands.Utility/commands/utility/Join-String.cs

```diff
@@ -160,79 +160,109 @@ protected override void EndProcessing()
         }
     }
 
-    [SuppressMessage(
-        "Microsoft.Performance",
-        "CA1812:AvoidUninstantiatedInternalClasses",
-        Justification = "Class is instantiated through late-bound reflection")]
-    internal class JoinItemCompleter : IArgumentCompleter
+    /// <summary>
+    /// Provides completion for the Separator parameter of the Join-String cmdlet.
+    /// </summary>
+    public class SeparatorArgumentCompleter : IArgumentCompleter
     {
-        public IEnumerable<CompletionResult> CompleteArgument(
-            string commandName,
-            string parameterName,
-            string wordToComplete,
-            CommandAst commandAst,
-            IDictionary fakeBoundParameters)
-        {
-            switch (parameterName)
-            {
-                case "Separator": return CompleteSeparator(wordToComplete);
-                case "FormatString": return CompleteFormatString(wordToComplete);
-            }
-
-            return null;
-        }
+        private static readonly string NewLineText =
+#if UNIX
+        "`n";
+#else
+        "`r`n";
+#endif
 
-        private static IEnumerable<CompletionResult> CompleteFormatString(string wordToComplete)
+        private static readonly IReadOnlyList<string> s_separatorValues = new List<string>
         {
-            var res = new List<CompletionResult>();
-            void AddMatching(string completionText)
-            {
-                if (completionText.StartsWith(wordToComplete, StringComparison.OrdinalIgnoreCase))
-                {
-                    res.Add(new CompletionResult(completionText));
-                }
-            }
-
-            AddMatching("'[{0}]'");
-            AddMatching("'{0:N2}'");
-            AddMatching("\"`r`n    `${0}\"");
-            AddMatching("\"`r`n    [string] `${0}\"");
+            ",",
+            ", ",
+            ";",
+            "; ",
+            NewLineText,
+            "-",
+            " ",
+        };
 
-            return res;
-        }
-
-        private IEnumerable<CompletionResult> CompleteSeparator(string wordToComplete)
+        private static readonly Dictionary<string, (string Tooltip, string ListItemText)> s_separatorMappings = new()
```

Comment:
```suggestion
        private static readonly Dictionary<string, (string Tooltip, string ListItemText)> s_separatorMappings = new(7)
```

Reply:
Added capacity here: 38bc0a7ee63c6bfe6be124a1f09311a012b726ee

And also instead of `KeyCollection` which doesn't work very well, I just set the keys to `IEnumerable<string>` in 5a2ab035e905a145df78b7d0064f00c3439b1b6e